integration/openshift/e2e/runner.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+set -e
+
+GOPATH=${GOPATH:-$(go env GOPATH)}
+script_dir="$(realpath $(dirname $0))"
+source "$script_dir/../../../lib/common.bash"
+tests_dir="$script_dir/../../../"
+source "$script_dir/../lib.sh"
+
+cmd="openshift-tests"
+readonly test_cases_file="${script_dir}/test_cases.txt"
+readonly default_kata_runtimeclass="$kata_runtimeclass"
+
+# Whether to run openshift-tests on dry-run mode or not.
+dry_run="false"
+# Whether to configure the OpenShift cluster or not.
+configure_cluster="false"
+
+function usage() {
+	cat << EOF
+This script runs a sub-set of the OpenShift e2e tests in other to test Kata
+Containers in a running OpenShift cluster.
+
+It assumes that the KUBECONFIG variable is exported and it points to a cluster
+where Kata Containers is already installed.
+
+The cluster also should have:
+ 1) The $default_kata_runtimeclass runtimeClass resource;
+ 2) A running admission controller which annotates any created Pod
+    to use that runtimeClass.
+
+If you choose to have the cluster configured by this script, beware that it
+will not return to its original state on the end.
+
+Usage $0: [-c] [-d] [-r NAME], where:
+  -h, --help   Print this message.
+  -c           Configure the cluster to run the tests.
+  -d           Run the tests in dry-run mode.
+  -r           The runtimeClass name (default is '$default_kata_runtimeclass').
+               This option should not be used with -c.
+EOF
+}
+
+[ -n "$KUBECONFIG" ] || die "The KUBECONFIG variable is unset"
+
+while getopts "hcdr:" opt; do
+	case ${opt} in
+		c)
+			configure_cluster="true"
+			;;
+		d)
+			dry_run="true"
+			;;
+		h)
+			usage
+			exit 0
+			;;
+		r)
+			kata_runtimeclass=${OPTARG}
+			;;
+		*)
+			usage
+			exit 1
+			;;
+	esac
+done
+
+# If the cluster is already configured by the user then only check that the
+# admission controller is active.
+#
+if [ "$configure_cluster" == "true" ]; then
+	[ "$default_kata_runtimeclass" == "$kata_runtimeclass" ] || \
+		die "To configure the cluster use the default runtimeClass"
+	configure_cluster || die "Failed to configure the cluster"
+else
+	info "Check the kata admission controller is active"
+	is_kata_admission_controller_active || \
+		die "Admission controller not active"
+fi
+
+cmd="openshift-tests"
+# Build the openshift tests binary if not available on PATH.
+if ! command -v "$cmd" >/dev/null; then
+	info "'$cmd' command not found on PATH. Build it from source"
+	cmd="$(build_openshift_tests)" || die "The build failed"
+fi
+cmd+=" run"
+
+# Save the tests results in junit format.
+#
+output_dir="${script_dir}/../../../_output/tests/openshift-e2e"
```

Comment:
Why all the path traversal? Could you do `output_dir=$(mktemp -d)` instead?

Reply:
I'm thinking on OpenShift CI when I set the output directory to `${tests_repo}/_output/tests/openshift-e2e`, because this way the generated junit file can be indexed properly. If I use a temp dir then the path is unpredictable to the caller CI scripts.

Reply:
Hmm. Maybe the caller could pass an "output direction parameter" to avoid this script having to make assumptions and use paths like this (which look confusing and error prone potentially to my eyes ;)

Reply:
I think you're right @jodh-intel, it deserves a parameter for the output in the `runner.sh`

Let me take that opportunity to ask one thing related... at begin I was attempted to use `KATA_TESTS_LOGDIR` (exported in `tests/lib/common.bash`) but I got the impression test scripts don't use it. Then the other day I saw in one of your PRs it is being used but only when `CI=true`. So I am curious about what is the orientation for the use of that variable (and others KATA_TESTS_*) in the scripts.

Reply:
That variable can be used by tests if needed. I think I added it for https://github.com/kata-containers/tests/blob/master/tracing/tracing-test.sh, but no other test really needs to use the feature since most just log to `stdout`/`stderr` and/or write to the journal.

Note that Jenkins collects up all these artifacts by calling the following at the end of a test run:

- https://github.com/kata-containers/tests/blob/master/.ci/teardown.sh